Test_Cases/JohnGemTest.java

```diff
@@ -0,0 +1,65 @@
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test cases for JohnGem class.
+ */
+public class JohnGemTest {
+
+	/**
+	 * JohnGem class instance
+	 */
+	JohnGem johnGemInstance;
+
+	/**
+	 * This method will execute before test case execution. initialization of
+	 * variable is done here
+	 */
+	@Before
+	public void setUp() {
+		johnGemInstance = new JohnGem();
+	}
+
+	/**
+	 * This method will execute after executing all the test cases. We are
+	 * nullifying the all the initialized variable.
+	 */
+	@After
+	public void tearDown() {
+		johnGemInstance = null;
+	}
+
+	/**
+	 * Test count gem positive.
+	 */
+	@Test
+	public void testFindNumberofGemElement() {
```

Comment:
Can you add more unit test case?
Also share the screen shot or unit test coverage?

Reply:
Added few more unit test cases.